drivers/clock_control/clock_control_nrf2_fll16m.c

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_fll16m
+
+#include "clock_control_nrf2_common.h"
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control/nrf_clock_control.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(clock_control_nrf2, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+#include <hal/nrf_lrcconf.h>
+
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
+	     "multiple instances not supported");
+
+#define FLAG_HFXO_STARTED BIT(FLAGS_COMMON_BITS)
+
+#define FLL16M_MODE_OPEN_LOOP   0
+#define FLL16M_MODE_CLOSED_LOOP 1
+#define FLL16M_MODE_BYPASS      2
+#define FLL16M_MODE_DEFAULT     FLL16M_MODE_OPEN_LOOP
+
+#define FLL16M_HAS_LFXO										\
+	(DT_INST_NODE_HAS_PROP(0, lfxo_clock) &&						\
+	 DT_NODE_HAS_STATUS(DT_INST_PROP(0, lfxo_clock), okay))
+
+#define FLL16M_LFXO_ACCURACY DT_PROP(DT_INST_PROP(0, lfxo_clock), accuracy_ppm)
+#define FLL16M_HFXO_ACCURACY DT_PROP(DT_INST_PROP(0, hfxo_clock), accuracy_ppm)
+#define FLL16M_OPEN_LOOP_ACCURACY DT_INST_PROP(0, open_loop_accuracy_ppm)
+#define FLL16M_CLOSED_LOOP_BASE_ACCURACY DT_INST_PROP(0, closed_loop_base_accuracy_ppm)
+#define FLL16M_MAX_ACCURACY FLL16M_HFXO_ACCURACY
+
+#if FLL16M_HAS_LFXO
+#define FLL16M_CLOSED_LOOP_ACCURACY (FLL16M_CLOSED_LOOP_BASE_ACCURACY + FLL16M_LFXO_ACCURACY)
+#else
+#define FLL16M_CLOSED_LOOP_ACCURACY (FLL16M_CLOSED_LOOP_BASE_ACCURACY + FLL16M_HFXO_ACCURACY)
+#endif
+
+/* Clock options sorted from lowest to highest accuracy */
+static const struct clock_options {
+	uint16_t accuracy;
+	uint8_t mode;
+} clock_options[] = {
+	{
+		.accuracy = FLL16M_OPEN_LOOP_ACCURACY,
+		.mode = FLL16M_MODE_OPEN_LOOP,
+	},
+	{
+		/* Closed-loop mode uses LFXO as source if present, HFXO otherwise */
```

Comment:
It seems this comment applies to lines 36-40 now

Reply:
moved to line 36